frontend/src/api/ApiCaller.tsx

```diff
@@ -134,25 +140,33 @@ export class BackendAPICaller {
         })
     }
 
-    async pauseMission(robotId: string) {
+    async pauseMission(robotId: string): Promise<ControlMissionResponse> {
         const path: string = 'robots/' + robotId + '/pause'
-        await this.POST(path, '').catch((e) => {
-            throw new Error(`Failed to POST /${path}: ` + e)
-        })
+        return this.postControlMissionRequest(path, robotId)
     }
 
-    async resumeMission(robotId: string) {
+    async resumeMission(robotId: string): Promise<ControlMissionResponse> {
         const path: string = 'robots/' + robotId + '/resume'
-        await this.POST(path, '').catch((e) => {
-            throw new Error(`Failed to POST /${path}: ` + e)
-        })
+        return this.postControlMissionRequest(path, robotId)
     }
 
-    async stopMission(robotId: string) {
+    async stopMission(robotId: string): Promise<ControlMissionResponse> {
         const path: string = 'robots/' + robotId + '/stop'
-        await this.POST(path, '').catch((e) => {
-            throw new Error(`Failed to POST /${path}: ` + e)
+        return this.postControlMissionRequest(path, robotId)
+    }
+
+    private async postControlMissionRequest(path: string, robotId: string): Promise<ControlMissionResponse> {
+        const url = `${config.BACKEND_URL}/${path}`
+        const body = { robotId: robotId }
+        const init: RequestInit = this.constructRequest('POST', body)
+
+        const response: Response = await fetch(url, init)
+        if (!response.ok) throw new Error(`${response.status} - ${response.statusText}`)
```

Comment:
Some duplicated code with the `query` function. Can this be abstracted out? May also help with the naming of `init`?